storage/docs/rest.md

```diff
@@ -1,69 +1,117 @@
 # REST Api
 
-## Store a file
+## URL scheme
+
+The REST Api uses the following scheme for urls
+
+``` /files/<namespace>/<key> ```
+
+The '<namespace>' component cannot include the the '/' character, while the '<key>' can include zero
+or more directories e.g:
+* ``` /files/libero/foo.txt ```
+* ``` /files/libero/baz/bar/foo.txt ```
+
+The key is unique within the namespace and provides a mapping between the storage entity and the provider entity (e.g. S3).
+
+## Storing a file
+
+To store a file with a namespace of ```libero``` and a key of ```directory/file.pdf```, the following request should be
+used
 
 ```
-PUT /files/namespace/directory/file.ext
+PUT /files/libero/directory/file.pdf
+
 Content-Type: application/pdf
 If-Match: <etag_value>|*
-Libero-file-tags: "key=value,key2=value2,filename=bar.pdf"
+Libero-file-tags: "filename=original-file.pdf,key1=value=1"
+```
 
+The Server should respond with:
+
+```
 Response:
 Code: 201
-Link: "<http://url/foo/bar.pdf>;rel=signed"
+Link: "<http://storage-url/path/to/file>;rel=public"
```

Comment:
`public` isn't a registered type (https://www.iana.org/assignments/link-relations/link-relations.xhtml) so it will need to be a URI (`http://libero.pub/public` or something; see https://tools.ietf.org/html/rfc8288#section-2.1.2).

Reply:
The link would actually be a direct link to the underlying storage url. So for s3 it would be something like:

https://mybucket.s3.amazonaws.com/directory/file.pdf

can we use the "original" type for example?

Reply:
`original` has a different meaning (original state of a resource).

Don't there there is one that describes this case (which is a publicly accessible version, so could be a CDN URI rather than a bucket directly).